### PR TITLE
ci: set up nyc coverage limits and coveralls action

### PR DESCRIPTION
**Pull request by boredland**
- uses coveralls github action to submit coverage
- sets nyc limits to realistic values
- adds a check how much a PR lowered the coverage

you can see it in action [here](https://github.com/boredland/bullmq/pull/14).

### CODE REVIEW

**Comment by boredland** [comment]
well well well. it seems coveralls isn't even set up? :joy:

**Comment by boredland** [comment]
in coveralls I'd recommend these settings:

image

in github I'd recommend these settings:

image

File: package.json [hunk 1]
```diff
@@ -120,7 +120,29 @@
     ],
     "exclude": [
       "bullmq-tests/test_*.ts"
-    ]
+    ],
+    "lines": 80,
```
Comment (manast) [inline package.json:124]: Are these settings related to coveralls?, I could not find documentation about them, are they supposed to be in the root of the package.json?

File: .github/workflows/coveralls.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests and submit coverage to coveralls
+# Since submitting the coverage needs the COVERALLS_REPO_TOKEN secret, this action runs in a dedicated pipeline
```
Comment (manast) [inline .github/workflows/coveralls.yml:2]: This comment does not seem to be up-to-date

Comment (boredland) [reply .github/workflows/coveralls.yml:2]: hi! I am currently working out a solution that allows to have a coverage feedback for guests like me. will keep you posted ;)

File: package.json (hunk 1 above)
Comment (boredland) [reply package.json:124]: No, this has nothing to do with coveralls, but with nyc (and isn't in the root, but inside the nyc field). For some time now, the bullmq repo has not been reporting to coveralls, because nyc has failed before any coveralls submission happened, because the default nyc threshold of 80% has been hit with the branch coverage being near 70%.
